components/LoginForm/LoginForm.jsx

```diff
@@ -58,7 +87,13 @@ export default function LoginForm() {
             Password
           </label>
           <input
-            onChange={(e) => setPassword(e.target.value)}
+            onChange={(e) =>
+              dispatch({
+                type: "input",
+                name: "password",
+                value: e.target.value,
+              })
+            }
```

Comment:
You can create a shared callback function instead of having to declare the dispatch in an anonymous function every time:

```jsx
const handleChange = (e) => {
  dispatch({
    type: "change", // currently called `input` but would make more sense as `change`
    name: e.target.name,
    value: e.target.value
  });
};

// ...

<label htmlFor="email">Email</label>
<input id="email" name="email" value={state.email} onChange={handleChange} />

<label htmlFor="password">Password</label>
<input id="password" name="password" value={state.password} onChange={handleChange} />
```